### PR TITLE
Set valid lifetime global option

### PR DESCRIPTION
**Pull request by jivdhaliwal**
# What
Adds the ability to set the valid lifetime (how long the addresses (leases) given out by the server are valid) as a global option

# Why
So that we can manage the valid lifetime of leases for all subnets

# Screenshots
ScreenShot 2020-10-23 at 12 16 53

# Notes
https://kea.readthedocs.io/en/kea-1.8.0/arm/dhcp4-srv.html#dhcpv4-server-configuration

### CODE REVIEW

File: app/models/global_option.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@ class GlobalOption < ApplicationRecord
     presence: {message: "must contain at least one IPv4 address separated using commas"},
     ipv4_list: {message: "contains an invalid IPv4 address or is not separated using commas"}
   validates :domain_name, presence: true
+  validates :valid_lifetime, numericality: {greater_than_or_equal_to: 0, only_integer: true},
```
Comment (neilkidd) [inline app/models/global_option.rb:9]: Is there a max value?

File: app/views/global_options/_form.html.erb [hunk 2]
```diff
@@ -3,28 +3,36 @@
               local: true do |f| %>
   <div class="govuk-form-group <%= field_error(f.object, :routers )%>">
     <%= f.label :routers, class: "govuk-label" %>
-    <div id="zone_forwarders-hint" class="govuk-hint">
+    <div id="global_option_routers-hint" class="govuk-hint">
```
Comment (neilkidd) [inline app/views/global_options/_form.html.erb:6]: 👍

**Review by neilkidd — APPROVED** [review]

File: app/models/global_option.rb (hunk 1 above)
Comment (jivdhaliwal) [reply app/models/global_option.rb:9]: Not that I could see. You can define your own max-valid-lifetime in the config but I don't see an upper limit when it is not set. It's looks like it's stored as a 32bit unsigned int so the max value would be the that

Comment (jivdhaliwal) [reply app/models/global_option.rb:9]: If someone tries to enter a value larger than a 32bit unsigned int (e.g. 4294967296), rails with throw an exception and prevent the change from reaching the kea config

**Pull request merged**